components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/utils/RestApiPublisherUtils.java

```diff
@@ -296,4 +298,29 @@ public static File exportOperationPolicyData(OperationPolicyData policyData)
             throw new APIManagementException("Error while exporting operation policy", e);
         }
     }
+
+    /**
+     * Validates the organization considering the API endpoint URL
+     * @param endpointUrl endpoint URL relevant of the API
+     * @param requestCtxOrdId organization ID related to the API invocation
+     * @return if the endpoint belongs to an authorized organization or not
+     * @throws APIManagementException happens if the Rudder returns 404 for the given namespace
+     */
+    public static boolean isEndpointBelongingToAuthorizedOrg(String endpointUrl, String requestCtxOrdId) throws APIManagementException {
+        if (endpointUrl.contains(APIConstants.SERVICE_ENDPOINT_URL_IDENTIFIER) &&
+                endpointUrl.split("\\.")[1] != null) {
+            String namespace = endpointUrl.split("\\.")[1];
+            try {
+                String orgIdFromNamespace = Rudder.getOrgIdFromNamespace(namespace);
+                return orgIdFromNamespace.equals(requestCtxOrdId);
+            } catch (FeignException e) {
+                if (e.status() == 404) {
+                    throw  new APIManagementException ("Could not find the organization given with the request",
+                            ExceptionCodes.ORGANIZATION_NOT_FOUND);
+                }
```

Comment:
How is this get handled? If a random URL given as the endpoint, are we throwing an error to the URL and fail the API creation?